src/Grid/grid.jl

```diff
@@ -726,7 +726,11 @@ end
 
 function Base.show(io::IO, ::MIME"text/plain", grid::Grid)
     print(io, "$(typeof(grid)) with $(getncells(grid)) ")
-    typestrs = sort!(collect(Set(repr(typeof(x)) for x in grid.cells)))
+    if isconcretetype(eltype(grid.cells))
+        typestrs = [repr(eltype(grid.cells))]
+    else
+        typestrs = sort!(repr.(Set(typeof(x) for x in grid.cells)))
+    end
```

Comment:
Can do something like this too, but feels a bit hacky..
```suggestion
    uniontypes(T::Union) = Set(getfield(T, key) for key in fieldnames(typeof(T)))
    cells = getcells(grid) # getcelltype(grid) N/A as it returns type of first cell
    CT = eltype(cells)
    if isa(CT, Union)
        typestrs = sort!(repr.(uniontypes(CT)))
    elseif isconcretetype(CT) || length(cells) > 10_000_000
        typestrs = [repr(CT)]
    else
        typestrs = sort!(repr.(Set(typeof(x) for x in cells)))
    end
```